dataset/test/dataset_test.cpp

```diff
@@ -225,6 +225,23 @@ TEST(DatasetTest, DataArrayView_setData) {
   EXPECT_EQ(d["a"].data(), var + var);
 }
 
+TEST(DatasetTest, size_in_memory_test) {
+  Dataset d;
+  const auto data =
+      makeVariable<double>(Dims{Dim::X}, Shape{3}, Values{1, 2, 3});
+  const auto mask = makeVariable<double>(Dims{Dim::X}, Shape{3});
+  const auto coords = makeVariable<double>(Dims{Dim::X}, Shape{3});
+
+  d.setData("a", data);
+  EXPECT_EQ(d.sizeInMemory(), sizeof(double) * 3);
```

Comment:
Test against the sizes of the variables, i.e., on the known correctness of `sizeInMemory` for `Variable`, to avoid "retesting" and extra math.